### PR TITLE
GH Actions and Tests Updates

### PR DESCRIPTION
**Pull request by prettycoder**
- update Dockerfiles for all types of linux to the newest Ruby patch versions
- adapt the travis matrix to the newest Ruby patch versions
- update GH actions and complete the workflow for all types of linux
- fix sidekiq and activejob related issues
- add fix to rails test so that the wicked binary can run on alpine
- adapt ruby_setup.sh and run_tests.sh scripts so that they run in GH as well as local docker containers

Tests don't pass 
Fixes coming shortly in future PR

### CODE REVIEW

File: .github/workflows/test_on_4_linux.yml [hunk 1]
```diff
@@ -1,27 +1,159 @@
...
+      rabbitmq:
+        image: rabbitmq:latest
+        ports:
+          - 5672:5672
+        options: --health-cmd "rabbitmqctl node_health_check" --health-interval 10s --health-timeout 5s --health-retries 5
+      mongo:
+        image: mongo:latest
+        ports:
+          - 27017:27017
+        options: "--health-cmd \"mongo --quiet --eval 'quit(db.runCommand({ ping: 1 }).ok ? 0 : 2)'\" --health-interval 10s --health-timeout 5s --health-retries 5"
```
Comment (ronilan) [inline .github/workflows/test_on_4_linux.yml:121]: I've been looking for that one long and wide. THANKS @prettycoder!

**Review by ronilan — APPROVED** [review]
I've left some comments for consideration, none of which I think should be blocking a merge.

File: test/run_tests/run_tests.sh [hunk 2]
```diff
@@ -57,16 +58,18 @@ The values for -r, -g, and -e have to correspond to configurations in the .travi
...
-    rm -rf /code/ruby-appoptics_test
-    cp -r /code/ruby-appoptics /code/ruby-appoptics_test
+    rm -rf /tmp/ruby-appoptics_test
+    cp -r . /tmp/ruby-appoptics_test
 
-    cd /code/ruby-appoptics_test/
+    cd /tmp/ruby-appoptics_test/
 fi
 
 ## Read travis configuration
```
[the remark this answers is not in the document]
Comment (prettycoder) [reply test/run_tests/run_tests.sh:75]: unfortunately I have no time for refactoring right now, there are a lot of travis references
created a ticket AO-20367

File: test/run_tests/Dockerfile_alpine [hunk 3]
```diff
@@ -99,12 +120,21 @@ RUN rc-update add memcached sysinit
...
+# install redis-server
+RUN curl -SL https://download.redis.io/releases/redis-6.2.4.tar.gz | tar xzC /tmp \
+    && cd /tmp/redis-6.2.4/ \
+    && make && make install && cd \
+    && rm -rf /tmp/redis-6.2.4/
+
 RUN apk add --update mysql-client mysql mariadb
 
+# postgres running in separate container and
+# this way of starting it doesn't seem to work anymore
```
[the remark this answers is not in the document]
Comment (prettycoder) [reply test/run_tests/Dockerfile_alpine:132]: I will once I know for sure I'm not going back to using it

File: .github/workflows/test_on_4_linux.yml [hunk 4]
```diff
@@ -1,27 +1,159 @@
...
+#        id:   check_files
+#        run:  |
+#          git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.sha }} ${{ github.sha }} | grep Dockerfile_
+#          echo ::set-output name=check_dockerfiles::$?
+##          git diff --name-only HEAD^ HEAD | grep [D]ockerfile_
+##          echo ::set-output name=check_dockerfiles::$?
+#
+#      - name: ghcr.io login ... build and publish images if needed
+#        uses: docker/login-action@v1
+#        if: ${{ steps.check_files.outputs.check_dockerfiles == 0}}
```
[the remark this answers is not in the document]
Comment (prettycoder) [reply .github/workflows/test_on_4_linux.yml:56]: until GH comes up with something usable ;)

File: .github/workflows/build_for_packagecloud.yml [hunk 5]
```diff
@@ -55,14 +55,16 @@ jobs:
...
-          APPOPTICS_SERVICE_KEY: ${{ secrets.APPOPTICS_SERVICE_KEY }}
-          APPOPTICS_COLLECTOR: ${{ secrets.APPOPTICS_COLLECTOR}}
-          APPOPTICS_FROM_S3: true
-        run: |
-          sleep 10
-          gem sources --add https://packagecloud.io/solarwinds/appoptics-apm-ruby/
-          gem install appoptics_apm --version ${{ steps.gemstep.outputs.GEM_VERSION }} --verbose
-          ruby ./scripts/test_install.rb
+# Run this test manually, github is having problems reading the index at packagecloud
+# TODO: AO-20355
```
[the remark this answers is not in the document]
Comment (prettycoder) [reply .github/workflows/build_for_packagecloud.yml:59]: still open

File: .github/workflows/run_tests.yml [hunk 6]
```diff
@@ -3,7 +3,10 @@
 
 name: Ruby Tests
 
-on: [push]
```
[the remark this answers is not in the document]
Comment (prettycoder) [reply .github/workflows/run_tests.yml:6]: thanks

File: test/run_tests/run_tests.sh [hunk 7]
```diff
@@ -17,8 +17,9 @@
 ##
```
[the remark this answers is not in the document]
Comment (prettycoder) [reply test/run_tests/run_tests.sh:17]: haha, let me run shellcheck ...

**Pull request merged**